rollup/da_syncer/blob_client/beacon_node_client.go

```diff
@@ -0,0 +1,183 @@
+package blob_client
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/crypto/kzg4844"
+	"github.com/scroll-tech/go-ethereum/rollup/rollup_sync_service"
+)
+
+type BeaconNodeClient struct {
+	apiEndpoint    string
+	l1Client       *rollup_sync_service.L1Client
+	genesisTime    uint64
+	secondsPerSlot uint64
+}
+
+var (
+	beaconNodeGenesisEndpoint = "/eth/v1/beacon/genesis"
+	beaconNodeSpecEndpoint    = "/eth/v1/config/spec"
+	beaconNodeBlobEndpoint    = "/eth/v1/beacon/blob_sidecars"
+)
+
+func NewBeaconNodeClient(apiEndpoint string, l1Client *rollup_sync_service.L1Client) (*BeaconNodeClient, error) {
+	// get genesis time
+	genesisPath, err := url.JoinPath(apiEndpoint, beaconNodeGenesisEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to join path, err: %w", err)
+	}
+	resp, err := http.Get(genesisPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot do request, err: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		bodyStr := string(body)
+		return nil, fmt.Errorf("beacon node request failed, status: %s, body: %s", resp.Status, bodyStr)
+	}
+
+	var genesisResp GenesisResp
+	err = json.NewDecoder(resp.Body).Decode(&genesisResp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode result into struct, err: %w", err)
+	}
+	genesisTime, err := strconv.ParseUint(genesisResp.Data.GenesisTime, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode genesis time %s, err: %w", genesisResp.Data.GenesisTime, err)
+	}
+
+	// get seconds per slot from spec
+	specPath, err := url.JoinPath(apiEndpoint, beaconNodeSpecEndpoint)
+	if err != nil {
+		return nil, fmt.Errorf("failed to join path, err: %w", err)
+	}
+	resp, err = http.Get(specPath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot do request, err: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		bodyStr := string(body)
+		return nil, fmt.Errorf("beacon node request failed, status: %s, body: %s", resp.Status, bodyStr)
+	}
+
+	var specResp SpecResp
+	err = json.NewDecoder(resp.Body).Decode(&specResp)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode result into struct, err: %w", err)
+	}
+	secondsPerSlot, err := strconv.ParseUint(specResp.Data.SecondsPerSlot, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode seconds per slot %s, err: %w", specResp.Data.SecondsPerSlot, err)
+	}
+	if secondsPerSlot == 0 {
```

Comment:
what's the rationale for this check?

Reply:
We use it later to divide to get the slot from the header time: https://github.com/scroll-tech/go-ethereum/blob/00494d58cf3390cbf909a06ef3ef46b0587d6d90/rollup/da_syncer/blob_client/beacon_node_client.go#L103

As such 0 is not allowed.